src/statistics.jl

```diff
@@ -0,0 +1,46 @@
+abstract type AbstractStatistic end
+
+struct Mean{D<:AbstractDomain} <: AbstractStatistic
+   domain::D
+   values::Dict{Symbol, Vector} 
+end
+
+Mean(domain, values) = Mean{typeof(domain)}(domain, values)
```

Comment:
I think we could eliminate this external constructor, and call the internal one in the constructor that takes a solution. See below.